src/recon_intra.rs

```diff
@@ -0,0 +1,434 @@
+];
+static has_tr_16x8: &'static [u8] = &[
+  255, 0, 85, 0, 119, 0, 85, 0, 127, 0, 85, 0, 119, 0, 85, 0,
+];
+static has_tr_16x16: &'static [u8] = &[
+  255, 85, 119, 85, 127, 85, 119, 85,
+];
+static has_tr_16x32: &'static [u8] = &[ 255, 119, 127, 119 ];
+static has_tr_32x16: &'static [u8] = &[ 15, 5, 7, 5 ];
+
+//pub static has_tr_32x32: [u8; 2] = [ 95, 87 ];
+static has_tr_32x32: &'static [u8] = &[ 95, 87 ];
+
+static has_tr_32x64: &'static [u8] = &[ 127 ];
+static has_tr_64x32: &'static [u8] = &[ 19 ];
+static has_tr_64x64: &'static [u8] = &[ 7 ];
+static has_tr_64x128: &'static [u8] = &[ 3 ];
+static has_tr_128x64: &'static [u8] = &[ 1 ];
+static has_tr_128x128: &'static [u8] = &[ 1 ];
+static has_tr_4x16: &'static [u8] = &[
+  255, 255, 255, 255, 127, 127, 127, 127, 255, 127, 255,
+  127, 127, 127, 127, 127, 255, 255, 255, 127, 127, 127,
+  127, 127, 255, 127, 255, 127, 127, 127, 127, 127,
+];
+static has_tr_16x4: &'static [u8] = &[
+  255, 0, 0, 0, 85, 0, 0, 0, 119, 0, 0, 0, 85, 0, 0, 0,
+  127, 0, 0, 0, 85, 0, 0, 0, 119, 0, 0, 0, 85, 0, 0, 0,
+];
+static has_tr_8x32: &'static [u8] = &[
+  255, 255, 127, 127, 255, 127, 127, 127,
+];
+static has_tr_32x8: &'static [u8] = &[
+  15, 0, 5, 0, 7, 0, 5, 0,
+];
+static has_tr_16x64: &'static [u8] = &[ 255, 127 ];
+static has_tr_64x16: &'static [u8] = &[ 3, 1 ];
+
+static has_tr_tables: &'static [&'static [u8]] = &[
+  // 4X4
+  has_tr_4x4,
+  // 4X8,       8X4,            8X8
+  has_tr_4x8, has_tr_8x4, has_tr_8x8,
+  // 8X16,      16X8,           16X16
+  has_tr_8x16, has_tr_16x8, has_tr_16x16,
+  // 16X32,     32X16,          32X32
+  has_tr_16x32, has_tr_32x16, has_tr_32x32,
+  // 32X64,     64X32,          64X64
+  has_tr_32x64, has_tr_64x32, has_tr_64x64,
+  // 64x128,    128x64,         128x128
+  has_tr_64x128, has_tr_128x64, has_tr_128x128,
+  // 4x16,      16x4,            8x32
+  has_tr_4x16, has_tr_16x4, has_tr_8x32,
+  // 32x8,      16x64,           64x16
+  has_tr_32x8, has_tr_16x64, has_tr_64x16
+];
+
+static has_tr_vert_8x8: &'static [u8] = &[
+  255, 255, 0, 0, 119, 119, 0, 0, 127, 127, 0, 0, 119, 119, 0, 0,
+  255, 127, 0, 0, 119, 119, 0, 0, 127, 127, 0, 0, 119, 119, 0, 0,
+];
+static has_tr_vert_16x16: &'static [u8] = &[
+  255, 0, 119, 0, 127, 0, 119, 0,
+];
+static has_tr_vert_32x32: &'static [u8] = &[ 15, 7 ];
+static has_tr_vert_64x64: &'static [u8] = &[ 3 ];
+
+// The _vert_* tables are like the ordinary tables above, but describe the
+// order we visit square blocks when doing a PARTITION_VERT_A or
+// PARTITION_VERT_B. This is the same order as normal except for on the last
+// split where we go vertically (TL, BL, TR, BR). We treat the rectangular block
+// as a pair of squares, which means that these tables work correctly for both
+// mixed vertical partition types.
+//
+// There are tables for each of the square sizes. Vertical rectangles (like
+// BLOCK_16X32) use their respective "non-vert" table
+/*
+static has_tr_vert_tables: &'static [&'static [u8]] = &[
```

Comment:
Do you still need this table?

Reply:
Once T shape partitions are added, they will be required.